src/Models/AdditionalParameters.php

```diff
@@ -33,6 +33,10 @@ class AdditionalParameters extends Model
      */
     public function setProperties(?array $data)
     {
+        if(!empty($data) && !isset($this->AdditionalParameter))
+        {
+            $this->AdditionalParameter = [];
+        }
```

Comment:
You can move this to the constructor. It will instantiate the object with a default empty array.

```suggestion
public function  __construct(?array $values = null)
    {
        $this->AdditionalParameter = $values ?? array();

        parent::__construct($values);
    }
```